### ISSUE TITLE
Milestones before the release of Vulcan 2

### ISSUE THREAD

**Issue by priyatharsan** [Aifred-Health/Vulcan2#18]
Date to complete: October 31th, 2018
@rfratila @nirtiac Feel free to add the what else been left to be done. Let's do this:

**Priya's tasks**
- [x] Change the criterion to object instance instead of name
  - [ ] Test manual custom criterion
- [x] Multi-input Network (https://arxiv.org/abs/1808.04456)
  - [x] Thoroughly test this once implemented
  - [x] Freeze network weights option
- [x] Other plot functionalities (@rfratila please add anything specific that we will need)
  - [x] t-sne
  - [x] pca
  - [x] receptive fields of a specific layer (most importantly the first layer)
  - [x] Metrics plot (e.g. Confusion Matrix, ROC curves - but these may already be implemented)
- [ ] Add more unittests to models (cnn, dnn, basenetwork - especially basenetwork)
- [x] Clean up examples folder (there are a few Jupiter notebooks that aren't super relevant like PyTorch_implementation.ipynb, PyTorch_train_function, and PyTorch_test_imports). Lets have different model configurations that show off the various ways to construct models.

**Caitrin's tasks**
- [ ] tests for dataset
- [x] implement unimplemented methods in dataset
- [x] cross-validation function
- [x] save function
  - [x] load function
- [x] tests for save and load
- [x] document metrics, layers and add tests at the same time

**Robert's tasks**
- [x] Saliency map plot
- [x] Learning rate scheduling
- [x] Snapshot ensemble
  - [x] train
  - [x] forward_pass
  - [x] save ensemble
  - [x] load ensemble

**Comment by priyatharsan** [comment]
@nirtiac the save and load are yet to be reconstructed, maybe we should add them under logger? what say @rfratila ?

**Comment by rfratila** [comment]
@priyatharsan, saving and loading should be placed where it is most intuitive. I would argue that they belong in basenetwork so that one can call `model.save_model(path=//)` and `model.load_model(path=//)`

**Comment by nirtiac** [comment]
@robert @priyatharsan  I agree, and with the classmethod!
